### PR TITLE
ShareWithMe icon in file list

### PR DESCRIPTION
**Pull request by purigarcia**
ShareWithMe icon is shown in the file list.
In the gridview, if it is a folder, shareWithMe icon is not shown. 
In the gridview, if it is a folder, ShareByLink is shown. The folder can be also be shared with me and It would be lose of information, because the icon above the folder is the ShareWithMe. If it is not correct and the icon should not be there, there is a commented line to be use.
In the gridview, if it is a file, both of them are shown

### CODE REVIEW

**Comment by davitol** [comment]
:+1: approved by QA

File: src/com/owncloud/android/ui/adapter/FileListListAdapter.java [hunk 1]
```diff
@@ -233,6 +233,7 @@ public View getView(int position, View convertView, ViewGroup parent) {
                 case GRID_IMAGE:
                     // sharedIcon
                     ImageView sharedIconV = (ImageView) view.findViewById(R.id.sharedIcon);
+                    //if (file.isShareByLink() && ((mGridMode && !file.isFolder()) || !mGridMode)) {
```
Comment (davivel) [inline src/com/owncloud/android/ui/adapter/FileListListAdapter.java]: Please, remove commented code.

File: src/com/owncloud/android/ui/adapter/FileListListAdapter.java [hunk 2]
```diff
@@ -268,17 +270,18 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
-                        if (checkIfFileIsSharedWithMe(file)) {
-                            sharedWithMeIconV.setVisibility(View.VISIBLE);
-                        } else {
-                            sharedWithMeIconV.setVisibility(View.GONE);
-                        }
+                    ImageView sharedWithMeIconV = (ImageView)
+                            view.findViewById(R.id.sharedWithMeIcon);
+                    sharedWithMeIconV.bringToFront();
+                    if (checkIfFileIsSharedWithMe(file) &&
+                            ((mGridMode && !file.isFolder()) || !mGridMode)) {
```
Comment (davivel) [inline src/com/owncloud/android/ui/adapter/FileListListAdapter.java]: This condition is equivalent to 
```(!file.isFolder() || !mGridMode)```

**Comment by davivel** [comment]
Reviewed; please, @purigarcia , check my comments

**Comment by davivel** [comment]
:+1:

**Pull request merged**